### PR TITLE
feat(lambda): support for Provisioned Pollers

### PR DESCRIPTION
**Pull request by roger-zhangg**
### Reason for this change

Lambda is introducing a new property in Event Sources named `ProvisionedPollerConfig` to set provisioned pollers that read from the event sources. When specified, it allows control over the minimum and maximum number of pollers that can be provisioned to process events from the source.

This feature is currently supported for MSK and Self-managed Kafka event sources.

### Description of changes

This new property can be opted in by setting the `ProvisionedPollerConfig` field while creating event sources. The example of setting `ProvisionedPollerConfig` for is shown below:

```
fn.addEventSource(new sources.ManagedKafkaEventSource(
  {
    clusterArn,
    topic: kafkaTopic,
    startingPosition: lambda.StartingPosition.TRIM_HORIZON,
    provisionedPollerConfig: {
      minimumPollers: 1,
      maximumPollers: 3,
    },
  }))

```
### Description of how you validated changes

Have added unit test and integration test to validate the implementation

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by xazhao — DISMISSED** [review]

File: packages/aws-cdk-lib/aws-lambda-event-sources/lib/stream.ts [hunk 1]
```diff
@@ -44,6 +44,29 @@ export interface BaseStreamEventSourceProps{
...
+   * @default - no provisioned pollers
+   */
+  readonly provisionedPollerConfig?: ProvisionedPollerConfig;
+}
+
+export interface ProvisionedPollerConfig {
+  /**
+   * The minimum number of pollers that should be provisioned.
+   *
+   * @default - 1
```
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-lambda-event-sources/lib/stream.ts:61]:
```suggestion
   * @default 1
```

File: packages/aws-cdk-lib/aws-lambda-event-sources/lib/stream.ts [hunk 2]
```diff
@@ -44,6 +44,29 @@ export interface BaseStreamEventSourceProps{
...
+   */
+  readonly minimumPollers: number;
+  /**
+   * The maximum number of pollers that can be provisioned.
+   *
+   * @default - 200
```
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-lambda-event-sources/lib/stream.ts:67]:
```suggestion
   * @default 200
```

**Review by GavinZZ — CHANGES_REQUESTED** [review]
nit comments

**Review by GavinZZ — DISMISSED** [review]

**Comment by GavinZZ** [comment]
@mergify update

File: packages/aws-cdk-lib/aws-lambda-event-sources/lib/stream.ts [hunk 3]
```diff
@@ -151,6 +174,24 @@ export interface StreamEventSourceProps extends BaseStreamEventSourceProps {
...
+        if (maximumPollers < 1 || maximumPollers > 2000) {
+          throw new Error('Maximum provisioned pollers must be between 1 and 2000 inclusive');
+        }
+      }
+      if (minimumPollers != undefined && maximumPollers != undefined) {
+        if (minimumPollers > maximumPollers) {
+          throw new Error('Minimum provisioned pollers must be less than or equal to maximum provisioned pollers');
+        }
+      }
+    }
```
Comment (moelasmar) [inline packages/aws-cdk-lib/aws-lambda-event-sources/lib/stream.ts:194]: could you please add unit test cases that cover these validations

**Review by GavinZZ — APPROVED** [review]

**Review by xazhao — APPROVED** [review]

**Pull request merged**